pkg/scheduler/schedulerdiscovery/ring_test.go

```diff
@@ -17,12 +18,12 @@ import (
 func TestRingConfig_DefaultConfigToBasicLifecyclerConfig(t *testing.T) {
 	cfg := RingConfig{}
 	flagext.DefaultValues(&cfg)
-	cfg.InstanceAddr = "127.0.0.1"
+	cfg.InstanceAddr = "::1"
 	cfg.InstancePort = 9095
 
 	expected := ring.BasicLifecyclerConfig{
 		ID:                              cfg.InstanceID,
-		Addr:                            fmt.Sprintf("%s:%d", cfg.InstanceAddr, cfg.InstancePort),
+		Addr:                            net.JoinHostPort(cfg.InstanceAddr, strconv.Itoa(cfg.InstancePort)),
```

Comment:
Rather than duplicating the logic between the production code and the test, should we specify the desired behaviour (eg. `Addr` is set to `"[::1]:9095"`) here?